### PR TITLE
feat: make app_id a public API

### LINKED ISSUE #127: Can we make Domino._app_id a public API?
Hi,

`Domino._app_id` is now 'protected', could it be a public API? We are making a domino product that needs to generate a URL to a deployed app, and for that, we need the app_id.
Happy to make a PR for this.

Regards,

Maarten Breddels

### PR DESCRIPTION
**Pull request by maartenbreddels**
Fixes #127

### CODE REVIEW

**Comment by maartenbreddels** [comment]
Fixes #128 

I decided to push two commits to this PR, since the seconds depends on the first.

**Comment by ddl-olsonJD** [comment]
Hey @maartenbreddels I will be looking into this. 
Do you also have some test cases for this?

**Comment by maartenbreddels** [comment]
Hi,

Happy to add them!
let me know what kind of test I could use as a template.

Regards,

Maarten

File: domino/domino.py [hunk 1]
```diff
@@ -806,24 +806,47 @@ def app_publish(self, unpublishRunningApps=True, hardwareTierId=None):
...
+        status = self.app_get_status(app_id)
         self.log.debug(f"App {app_id} status={status}")
         if status and status != "Stopped" and status != "Failed":
             url = self._routes.app_stop(app_id)
             response = self.request_manager.post(url)
             return response
 
-    def __app_get_status(self, id) -> Optional[str]:
-        app_id = self._app_id
+    def app_id(self):
```
Comment (ddl-olsonJD) [inline domino/domino.py:819]: Set as a property as well.

File: domino/domino.py [hunk 2]
```diff
@@ -806,24 +806,47 @@ def app_publish(self, unpublishRunningApps=True, hardwareTierId=None):
...
+        else:
+            app_id = None
+        return app_id
+
+    @property
+    def _app_id(self):
+        # for backwards compatibility, we keep this property
+        return self.app_id()
+
+    def app_get_status(self, id) -> Optional[str]:
```
Comment (ddl-olsonJD) [inline domino/domino.py:838]:
```
def def app_get_status(self, ext_app_id=None) -> Optional[str]:
    app_id = ext_app_id if ext_app_id else self.app_id
```

File: domino/domino.py [hunk 3]
```diff
@@ -806,24 +806,47 @@ def app_publish(self, unpublishRunningApps=True, hardwareTierId=None):
...
+        app_id = self.app_id()
         if app_id is None:
             return None
         url = self._routes.app_get(app_id)
         response = self.request_manager.get(url).json()
         return response.get("status", None)
 
+    def __app_get_status(self, id) -> Optional[str]:
+        # for backwards compatibility, we keep this method
+        return self.__app_get_status()
```
Comment (ddl-olsonJD) [inline domino/domino.py:848]: meant to call `app_get_status`?

**Review by ddl-olsonJD — CHANGES_REQUESTED** [review]
And for tests, you can follow the test suites that we already have, but around app publishing and un-publishing

**Comment by ddl-olsonJD** [comment]
[Jira Ticket - DOM-43074](https://dominodatalab.atlassian.net/browse/DOM-43074)

**Comment by ddl-olsonJD** [comment]
close as out of date

**Pull request closed without merging**